kv/kvserver: TestReplicateQueueRebalanceMultiStore failed

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17365645?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17365645?buildTab=artifacts#/) on master @ [3eb2bb04abaaaaac92f3f5f2f6952a30ada78de5](https://github.com/cockroachdb/cockroach/commits/3eb2bb04abaaaaac92f3f5f2f6952a30ada78de5):


```
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [54 6 7 6 6]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [53 6 7 7 6]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [52 6 7 8 6]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [51 6 7 9 6]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [50 7 7 9 6]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [49 9 7 8 6]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [48 10 7 7 7]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [47 10 7 6 9]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [46 10 7 6 10]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [45 9 8 8 9]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [44 8 10 8 9]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [43 7 11 8 10]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [42 7 10 9 11]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [41 7 9 10 12]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [40 7 8 12 12]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [39 6 9 13 12]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [38 7 11 12 11]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [37 10 10 11 11]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [36 11 9 10 13]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [35 10 12 10 12]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [34 9 13 12 11]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [33 10 12 14 10]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [32 13 11 13 10]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [31 15 10 12 11]
    replicate_queue_test.go:334: current replica state (want at least 67 replicas on all stores): [74 75 73 77 74]
    replicate_queue_test.go:345: current lease state (want at least 11 leases on all stores): [30 14 12 11 12]
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-43338

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/94664c35-10a6-4f48-a11b-30ecf1984958?testReportRun=5&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [9c3acd84e2fc639328b27a6cacd0e3893f1f90bb](https://github.com/cockroachdb/cockroach/commits/9c3acd84e2fc639328b27a6cacd0e3893f1f90bb):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x142e811]

```
Stack: 

```
goroutine 1537181 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc012fdb868?, {0x8d99218, 0xc00dfc07e0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x6895420?, 0xd0f5aa0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc012fdb988?, {0x8d99218, 0xc00dfc0960})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x6895420?, 0xd0f5aa0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit(0x0, {0x8d99218, 0xc00dfc0fc0}, 0x1142)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165 +0x71
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace(0xc012fdc768, {0x8d99218?, 0xc00dfc0fc0?}, 0xc00635dce0?, 0x18?)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292 +0x6d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive(_, {_, _}, _, {_, _}, {{0x11, 0x5, 0xc007914a10, 0xc007914a80, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918 +0x1f9f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot(0xc00ef3c008, {0x8d99218, 0xc00dfc0960}, 0xc007095208, {0x7fe6eb9ee3b0, 0xc002243e30})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693 +0x76b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1({0x8d99218, 0xc00dfc0960})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209 +0x85
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc0075cbd40, {0x8d99218, 0xc00dfc0960}, {0x746e8ec, 0x1e}, 0xc00b2f1280)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot(0xc00ef3c008, {0x8d991e0?, 0xc001a60500?}, 0xc007095208, {0x7fe6eb9ee388, 0xc002243e30})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206 +0xf6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot(0xc00de3c380, {0x8de74e8, 0xc002243e30})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638 +0x32d
github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler({0x72a7fc0?, 0xc00de3c380}, {0x8dd19d0, 0xc0046e6000})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267 +0xd8
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13({0x72a7fc0, 0xc00de3c380}, {0x8dd1748, 0xc00dbaea50}, 0xc01324e108, 0x787ebd8)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160 +0x35b
google.golang.org/grpc.getChainStreamHandler.func1({0x72a7fc0, 0xc00de3c380}, {0x8dd1748, 0xc00dbaea50})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x72a7fc0, 0xc00de3c380}, {0x8dd1748, 0xc00dbaea50}, 0xc01324e108?, 0xc0070bfc80)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174 +0x6c
google.golang.org/grpc.getChainStreamHandler.func1({0x72a7fc0, 0xc00de3c380}, {0x8dd1748, 0xc00dbaea50})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc008764000?, {{0x470445?}, {0x8df41a0?, 0xc00ef6f740?}}}, {0x72a7fc0, 0xc00de3c380}, {0x8dd1748, 0xc00dbaea50}, 0xc01324e108, 0xc0070bfc40)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152 +0x384
google.golang.org/grpc.getChainStreamHandler.func1({0x72a7fc0, 0xc00de3c380}, {0x8dd1748, 0xc00dbaea50})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x7fe6ac1b7a58?, 0x8d99218?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141 +0x27
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc0075cbd40, {0x8d99218, 0xc00dfc07e0}, {0xc01427cd80, 0x29}, 0xc00af6aaa0)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x72a7fc0, 0xc00de3c380}, {0x8dd1748, 0xc00dbaea50}, 0xc01324e108, 0xc0070bfc00)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140 +0xdf
google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2({0x72a7fc0, 0xc00de3c380}, {0x8dd1748, 0xc00dbaea50}, 0xc01324e108, 0xc002243de0?)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472 +0x85
google.golang.org/grpc.(*Server).processStreamingRPC(0xc004e5ed20, {0x8df5940, 0xc012a3dd40}, 0xc007f939e0, 0xc0170c76e0, 0xd1f2240, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636 +0x1247
google.golang.org/grpc.(*Server).handleStream(0xc004e5ed20, {0x8df5940, 0xc012a3dd40}, 0xc007f939e0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716 +0x999
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 1537205
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:958 +0x136
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140
*   | google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472
*   | google.golang.org/grpc.(*Server).processStreamingRPC
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636
*   | google.golang.org/grpc.(*Server).handleStream
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716
*   | google.golang.org/grpc.(*Server).serveStreams.func1.1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/f9978e6b-8d20-4dcd-aeaa-975a39247769?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x142fcd1]

```
Stack: 

```
goroutine 808251 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc00ac9b820?, {0x8db15f8, 0xc009a26900})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x68a8500?, 0xd11caa0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc00ac9b940?, {0x8db15f8, 0xc00797f170})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x68a8500?, 0xd11caa0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit(0x0, {0x8db15f8, 0xc00797fbc0}, 0x557)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165 +0x71
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace(0xc00ac9c720, {0x8db15f8?, 0xc00797fbc0?}, 0xc011448840?, 0xf8?)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292 +0x6d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive(_, {_, _}, _, {_, _}, {{0xe, 0x2, 0xc0112e6770, 0xc0112e67e0, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918 +0x1f9f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot(0xc009eac808, {0x8db15f8, 0xc00797f170}, 0xc00f8e4488, {0x7f7ad0671828, 0xc012deb350})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693 +0x797
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1({0x8db15f8, 0xc00797f170})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209 +0x85
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00edea120, {0x8db15f8, 0xc00797f170}, {0x7483b44, 0x1e}, 0xc002db5268)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot(0xc009eac808, {0x8db15c0?, 0xc01433d0e0?}, 0xc00f8e4488, {0x7f7ad0671800, 0xc012deb350})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206 +0xf6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot(0xc002899a40, {0x8dff768, 0xc012deb350})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638 +0x30d
github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler({0x72bcd60?, 0xc002899a40}, {0x8de9d30, 0xc007c2bd00})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267 +0xd8
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13({0x72bcd60, 0xc002899a40}, {0x8de9aa8, 0xc009550780}, 0xc006a9ff50, 0x78943a8)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160 +0x35b
google.golang.org/grpc.getChainStreamHandler.func1({0x72bcd60, 0xc002899a40}, {0x8de9aa8, 0xc009550780})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x72bcd60, 0xc002899a40}, {0x8de9aa8, 0xc009550780}, 0xc006a9ff50?, 0xc0010a7080)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174 +0x6c
google.golang.org/grpc.getChainStreamHandler.func1({0x72bcd60, 0xc002899a40}, {0x8de9aa8, 0xc009550780})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc00ef86000?, {{0x470445?}, {0x8e0c420?, 0xc01ba79d40?}}}, {0x72bcd60, 0xc002899a40}, {0x8de9aa8, 0xc009550780}, 0xc006a9ff50, 0xc0010a6fc0)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152 +0x384
google.golang.org/grpc.getChainStreamHandler.func1({0x72bcd60, 0xc002899a40}, {0x8de9aa8, 0xc009550780})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x7f7ad43815f8?, 0x8db15f8?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141 +0x27
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00edea120, {0x8db15f8, 0xc009a26900}, {0xc00ac583c0, 0x29}, 0xc0083f8aa0)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x72bcd60, 0xc002899a40}, {0x8de9aa8, 0xc009550780}, 0xc006a9ff50, 0xc0010a6f80)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140 +0xdf
google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2({0x72bcd60, 0xc002899a40}, {0x8de9aa8, 0xc009550780}, 0xc006a9ff50, 0xc012deb340?)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472 +0x85
google.golang.org/grpc.(*Server).processStreamingRPC(0xc0049f3a40, {0x8e0dbc0, 0xc006a431e0}, 0xc0120db0e0, 0xc00c2726c0, 0xd2192e0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636 +0x1247
google.golang.org/grpc.(*Server).handleStream(0xc0049f3a40, {0x8e0dbc0, 0xc006a431e0}, 0xc0120db0e0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716 +0x999
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 808166
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:958 +0x136
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140
*   | google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472
*   | google.golang.org/grpc.(*Server).processStreamingRPC
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636
*   | google.golang.org/grpc.(*Server).handleStream
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716
*   | google.golang.org/grpc.(*Server).serveStreams.func1.1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #132995 kv/kvserver: TestReplicateQueueRebalanceMultiStore failed [C-test-failure O-robot T-kv branch-release-24.3 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/b86ff27e-b525-4113-ad6c-0db94f54a0d1?testReportRun=17&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x142fcd1]

```
Stack: 

```
goroutine 508705 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc010d07820?, {0x8db15f8, 0xc018fee5a0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x68a8500?, 0xd11caa0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc010d07940?, {0x8db15f8, 0xc018fee6f0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x68a8500?, 0xd11caa0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit(0x0, {0x8db15f8, 0xc018feecf0}, 0x557)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165 +0x71
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace(0xc010d08720, {0x8db15f8?, 0xc018feecf0?}, 0xc0111bc2c0?, 0xf8?)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292 +0x6d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive(_, {_, _}, _, {_, _}, {{0xe, 0x2, 0xc009a047e0, 0xc009a04850, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918 +0x1f9f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot(0xc01206cc08, {0x8db15f8, 0xc018fee6f0}, 0xc012c1d208, {0x7f44d9a10f80, 0xc01be3af70})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693 +0x797
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1({0x8db15f8, 0xc018fee6f0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209 +0x85
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00f7407e0, {0x8db15f8, 0xc018fee6f0}, {0x7483b44, 0x1e}, 0xc0058db268)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot(0xc01206cc08, {0x8db15c0?, 0xc012412230?}, 0xc012c1d208, {0x7f44d9a10f58, 0xc01be3af70})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206 +0xf6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot(0xc00f757180, {0x8dff768, 0xc01be3af70})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638 +0x30d
github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler({0x72bcd60?, 0xc00f757180}, {0x8de9d30, 0xc012a373e0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267 +0xd8
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13({0x72bcd60, 0xc00f757180}, {0x8de9aa8, 0xc010c6b0e0}, 0xc015f01848, 0x78943a8)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160 +0x35b
google.golang.org/grpc.getChainStreamHandler.func1({0x72bcd60, 0xc00f757180}, {0x8de9aa8, 0xc010c6b0e0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x72bcd60, 0xc00f757180}, {0x8de9aa8, 0xc010c6b0e0}, 0xc015f01848?, 0xc00bf6c200)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174 +0x6c
google.golang.org/grpc.getChainStreamHandler.func1({0x72bcd60, 0xc00f757180}, {0x8de9aa8, 0xc010c6b0e0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc00fbce000?, {{0x470445?}, {0x8e0c420?, 0xc00ef97800?}}}, {0x72bcd60, 0xc00f757180}, {0x8de9aa8, 0xc010c6b0e0}, 0xc015f01848, 0xc00bf6c1c0)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152 +0x384
google.golang.org/grpc.getChainStreamHandler.func1({0x72bcd60, 0xc00f757180}, {0x8de9aa8, 0xc010c6b0e0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x7f44d9a2d478?, 0x8db15f8?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141 +0x27
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00f7407e0, {0x8db15f8, 0xc018fee5a0}, {0xc00947b3e0, 0x29}, 0xc00b4bbaa0)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x72bcd60, 0xc00f757180}, {0x8de9aa8, 0xc010c6b0e0}, 0xc015f01848, 0xc00bf6c180)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140 +0xdf
google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2({0x72bcd60, 0xc00f757180}, {0x8de9aa8, 0xc010c6b0e0}, 0xc015f01848, 0xc01be3af60?)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472 +0x85
google.golang.org/grpc.(*Server).processStreamingRPC(0xc0137cba40, {0x8e0dbc0, 0xc00befd1e0}, 0xc00f24f9e0, 0xc0016eaa80, 0xd2192e0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636 +0x1247
google.golang.org/grpc.(*Server).handleStream(0xc0137cba40, {0x8e0dbc0, 0xc00befd1e0}, 0xc00f24f9e0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716 +0x999
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 508708
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:958 +0x136
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140
*   | google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472
*   | google.golang.org/grpc.(*Server).processStreamingRPC
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636
*   | google.golang.org/grpc.(*Server).handleStream
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716
*   | google.golang.org/grpc.(*Server).serveStreams.func1.1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=17</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #132995 kv/kvserver: TestReplicateQueueRebalanceMultiStore failed [C-test-failure O-robot T-kv branch-release-24.3 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/fa614228-f8fe-4e0c-99f1-fcd3dd9c8417?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x142fcd1]

```
Stack: 

```
goroutine 1578744 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc012497820?, {0x8db15f8, 0xc0136de5d0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x68a8500?, 0xd11caa0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc012497940?, {0x8db15f8, 0xc0136de750})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x68a8500?, 0xd11caa0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit(0x0, {0x8db15f8, 0xc0136de840}, 0x111e)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165 +0x71
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace(0xc012498720, {0x8db15f8?, 0xc0136de840?}, 0xc00e17f1e0?, 0xf8?)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292 +0x6d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive(_, {_, _}, _, {_, _}, {{0x12, 0x6, 0xc00837e310, 0xc00837e380, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918 +0x1f9f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot(0xc00ec2e008, {0x8db15f8, 0xc0136de750}, 0xc00837ab48, {0x7f0dfaba8728, 0xc001c1c4e0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693 +0x797
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1({0x8db15f8, 0xc0136de750})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209 +0x85
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00365d830, {0x8db15f8, 0xc0136de750}, {0x7483b44, 0x1e}, 0xc00a825268)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot(0xc00ec2e008, {0x8db15c0?, 0xc001109a90?}, 0xc00837ab48, {0x7f0dfaba8700, 0xc001c1c4e0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206 +0xf6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot(0xc00e1b0540, {0x8dff768, 0xc001c1c4e0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638 +0x30d
github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler({0x72bcd60?, 0xc00e1b0540}, {0x8de9d30, 0xc00f9d8fc0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267 +0xd8
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13({0x72bcd60, 0xc00e1b0540}, {0x8de9aa8, 0xc010d16780}, 0xc00fb45878, 0x78943a8)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160 +0x35b
google.golang.org/grpc.getChainStreamHandler.func1({0x72bcd60, 0xc00e1b0540}, {0x8de9aa8, 0xc010d16780})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x72bcd60, 0xc00e1b0540}, {0x8de9aa8, 0xc010d16780}, 0xc00fb45878?, 0xc0086bc4c0)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174 +0x6c
google.golang.org/grpc.getChainStreamHandler.func1({0x72bcd60, 0xc00e1b0540}, {0x8de9aa8, 0xc010d16780})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc007d38000?, {{0x470445?}, {0x8e0c420?, 0xc00cf09e00?}}}, {0x72bcd60, 0xc00e1b0540}, {0x8de9aa8, 0xc010d16780}, 0xc00fb45878, 0xc0086bc480)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152 +0x384
google.golang.org/grpc.getChainStreamHandler.func1({0x72bcd60, 0xc00e1b0540}, {0x8de9aa8, 0xc010d16780})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x7f0dfb88e348?, 0x8db15f8?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141 +0x27
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00365d830, {0x8db15f8, 0xc0136de5d0}, {0xc00e9bf860, 0x29}, 0xc00ac90aa0)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x72bcd60, 0xc00e1b0540}, {0x8de9aa8, 0xc010d16780}, 0xc00fb45878, 0xc0086bc440)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140 +0xdf
google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2({0x72bcd60, 0xc00e1b0540}, {0x8de9aa8, 0xc010d16780}, 0xc00fb45878, 0xc001c1c4d0?)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472 +0x85
google.golang.org/grpc.(*Server).processStreamingRPC(0xc006b3a960, {0x8e0dbc0, 0xc005e596c0}, 0xc0109a98c0, 0xc018217bf0, 0xd2192e0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636 +0x1247
google.golang.org/grpc.(*Server).handleStream(0xc006b3a960, {0x8e0dbc0, 0xc005e596c0}, 0xc0109a98c0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716 +0x999
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 1578656
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:958 +0x136
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140
*   | google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472
*   | google.golang.org/grpc.(*Server).processStreamingRPC
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636
*   | google.golang.org/grpc.(*Server).handleStream
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716
*   | google.golang.org/grpc.(*Server).serveStreams.func1.1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #132995 kv/kvserver: TestReplicateQueueRebalanceMultiStore failed [C-test-failure O-robot T-kv branch-release-24.3 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/kv/kvserver/store_snapshot.go
+++ b/pkg/kv/kvserver/store_snapshot.go
@@ -735,13 +735,14 @@ func (kvSS *kvBatchSnapshotStrategy) Receive(
 	var prevWriteBytes int64
 
 	snapshotQ := s.cfg.KVAdmissionController.GetSnapshotQueue(s.StoreID())
+	if snapshotQ == nil {
+		log.Errorf(ctx, "unable to find snapshot queue for store: %s", s.StoreID())
+	}
 	// Using a nil pacer is effectively a noop if snapshot control is disabled.
 	var pacer *admission.SnapshotPacer = nil
-	if admission.DiskBandwidthForSnapshotIngest.Get(&s.cfg.Settings.SV) {
-		pacer = admission.NewSnapshotPacer(snapshotQ, s.StoreID())
+	if admission.DiskBandwidthForSnapshotIngest.Get(&s.cfg.Settings.SV) && snapshotQ != nil {
+		pacer = admission.NewSnapshotPacer(snapshotQ)
 	}
-	// It is safe to call Close() on a nil pacer.
-	defer pacer.Close()
 
 	for {
 		timingTag.start("recv")

--- a/pkg/util/admission/snapshot_queue.go
+++ b/pkg/util/admission/snapshot_queue.go
@@ -11,9 +11,7 @@ import (
 	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
-	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings"
-	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/metamorphic"
 	"github.com/cockroachdb/cockroach/pkg/util/metric"
 	"github.com/cockroachdb/cockroach/pkg/util/queue"
@@ -263,28 +261,21 @@ func newSnapshotWorkItem(count int64) *snapshotWorkItem {
 
 type SnapshotPacer struct {
 	snapshotQ     snapshotRequester
-	storeID       roachpb.StoreID
 	intWriteBytes int64
 }
 
-func NewSnapshotPacer(q snapshotRequester, s roachpb.StoreID) *SnapshotPacer {
+func NewSnapshotPacer(q snapshotRequester) *SnapshotPacer {
 	return &SnapshotPacer{
 		snapshotQ:     q,
-		storeID:       s,
 		intWriteBytes: 0,
 	}
 }
 
 func (p *SnapshotPacer) Pace(ctx context.Context, writeBytes int64, final bool) error {
-	// Return early if nil pacer or nil snapshotQ.
+	// Return early if nil pacer.
 	if p == nil {
 		return nil
 	}
-	if p.snapshotQ == nil {
-		log.Errorf(ctx, "unable to find snapshot queue for store: %s", p.storeID)
-		return nil
-	}
-
 	p.intWriteBytes += writeBytes
 	if p.intWriteBytes <= SnapshotBurstSize && !final {
 		return nil
@@ -294,11 +285,4 @@ func (p *SnapshotPacer) Pace(ctx context.Context, writeBytes int64, final bool) 
 	}
 	p.intWriteBytes = 0
 	return nil
-}
-
-func (p *SnapshotPacer) Close() {
-	if p == nil {
-		return
-	}
-	p.snapshotQ = nil
 }

--- a/pkg/util/admission/snapshot_queue_test.go
+++ b/pkg/util/admission/snapshot_queue_test.go
@@ -136,12 +136,9 @@ func TestSnapshotPacer(t *testing.T) {
 	var pacer *SnapshotPacer = nil
 	// Should not panic on nil pacer.
 	require.NoError(t, pacer.Pace(ctx, 1, false))
-	// Should not panic on nil pacer.
-	pacer.Close()
 
 	q := &testingSnapshotQueue{}
-	pacer = NewSnapshotPacer(q, 1)
-	defer pacer.Close()
+	pacer = NewSnapshotPacer(q)
 
 	// Should not ask for admission since write bytes = burst size.
 	writeBytes := int64(SnapshotBurstSize)

